src/components/ha-control-switch.ts

```diff
@@ -53,14 +63,68 @@ export class HaControlSwitch extends LitElement {
 
   connectedCallback(): void {
     super.connectedCallback();
-    this.addEventListener("keydown", this._keydown);
-    this.addEventListener("click", this._toggle);
+    this.setupListeners();
   }
 
   disconnectedCallback(): void {
     super.disconnectedCallback();
+    this.destroyListeners();
+  }
+
+  @query("#switch")
+  private switch;
+
+  setupListeners() {
+    if (this.switch && !this._mc) {
+      this._mc = new Manager(this.switch, {
+        touchAction: this.vertical ? "pan-x" : "pan-y",
+      });
+      this._mc.add(
+        new Swipe({
+          direction: this.vertical ? DIRECTION_VERTICAL : DIRECTION_HORIZONTAL,
+        })
+      );
+
+      this._mc.add(new Tap({ event: "singletap" }));
+
+      this._mc.on("swipeup", () => {
+        if (this.disabled) return;
```

Comment:
Should we even add listeners when disabled?

Reply:
It avoid adding/removing them when disabled props change. I agree we can improve this ! 🙂